### PR TITLE
[2.0] Continue GLIDE scans across empty batches

### LINKED ISSUE #108: [2.0] GLIDE standalone SCAN stops on an empty intermediate batch
## Bug

The standalone GLIDE implementation of `Cursor.hasNext()` in Spring Data Valkey 2.0 stops a `SCAN` iteration when one response contains no matching keys, even if the response's continuation cursor is non-zero.

Valkey permits `SCAN MATCH` to return an empty collection before iteration is complete. Completion is indicated only by cursor `0`.

## Protocol and driver documentation

- The official [Valkey SCAN documentation](https://valkey.io/commands/scan/#number-of-elements-returned-at-every-scan-call) explicitly states that a call may return zero elements and that clients must continue while the returned cursor is non-zero. Its `MATCH` example also shows several empty responses with non-zero cursors before a later response returns matches.
- The [Valkey GLIDE migration guide's SCAN example](https://github.com/valkey-io/valkey-glide/wiki/Migration-Guide-Lettuce#keys--scan) follows the same contract: it processes a batch only when keys are present, but terminates iteration based solely on cursor `0`.

## Affected version

- `io.valkey.springframework.data:spring-data-valkey:2.0.0`
- GLIDE standalone connection

The defect was verified directly against the released `v2.0.0` tag. At the time of verification, `v2.0.0` and `main` point to commit `427208e36e8874173a376639c49a3e3017f716ea`, and both contain the same faulty cursor implementation. This issue and the proposed fix target the 2.0 code line.

## Reproduction

Create 32 matching keys and scan with a small count hint:

```java
for (int i = 0; i < 32; i++) {
    template.opsForValue().set("scan:" + i, "value");
}

try (Cursor<String> cursor = template.scan(
        ScanOptions.scanOptions().match("scan:*").count(1).build())) {
    List<String> keys = new ArrayList<>();
    cursor.forEachRemaining(keys::add);
    assertThat(keys).hasSize(32);
}
```

The assertion fails intermittently with only a subset of keys.

## Cause

`ValkeyGlideKeyCommands.ValkeyGlideKeyScanCursor.hasNext()` calls `loadNextBatch()` once and then returns whether that batch contains an item. If the batch is empty, it returns `false` even when `loadNextBatch()` updated the cursor to a non-zero value.

## Expected behavior

`hasNext()` should keep loading empty intermediate batches until either:

- a batch contains an item, or
- the server returns cursor `0`.

This is analogous to the empty-iteration behavior previously tracked in spring-projects/spring-data-redis#995 (`DATAREDIS-417`).

### PR DESCRIPTION
**Pull request by progys**
Closes #108.

## Target

This fixes Spring Data Valkey `2.0.0`. The defect was reproduced in the released `v2.0.0` tag, and this branch is based on `main`, which currently points to the same commit as that tag.

## Protocol and driver documentation

- The official [Valkey SCAN documentation](https://valkey.io/commands/scan/#number-of-elements-returned-at-every-scan-call) explicitly states that a call may return zero elements and that clients must continue while the returned cursor is non-zero. Its `MATCH` example also shows empty intermediate responses.
- The [Valkey GLIDE migration guide's SCAN example](https://github.com/valkey-io/valkey-glide/wiki/Migration-Guide-Lettuce#keys--scan) processes keys only when a batch is non-empty but continues until cursor `0`.

## Change

An empty SCAN result does not mean iteration is complete when the returned cursor is non-zero. The GLIDE standalone cursor previously loaded only one batch per `hasNext()` call and returned `false` if that batch was empty, causing callers to stop before later matches were returned.

This change makes `hasNext()` keep loading batches until it finds a key or receives cursor zero. It also adds a deterministic unit test covering a non-empty batch, an empty intermediate batch with a non-zero cursor, and a final non-empty batch. The cursor Javadoc documents why empty intermediate batches must be skipped and links directly to the Valkey SCAN contract.

- [x] I have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] I used the project code format and did not include unrelated formatting changes.
- [x] I submitted a test case that backs the change.
- [x] I added myself as author in the headers of the classes I touched.

## Tests

- `./mvnw test -pl spring-data-valkey -Dtest=ValkeyGlideKeyCommandsUnitTests` — passes (1 test).
- `make test` — the new test passes, but the full suite fails in unrelated existing tests because the harness builds Valkey 8.1.1 while tests invoke commands not supported there, including `HSETEX`, `HEXPIREAT`, and `XDELEX` (28,706 run; 51 failures; 130 errors; 613 skipped).

### CODE REVIEW

File: spring-data-valkey/src/test/java/io/valkey/springframework/data/valkey/connection/valkeyglide/ValkeyGlideKeyCommandsUnitTests.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2026-present the original author or authors.
```
[the remark this answers is not in the document]
Comment (progys) [reply spring-data-valkey/src/test/java/io/valkey/springframework/data/valkey/connection/valkeyglide/ValkeyGlideKeyCommandsUnitTests.java]: Fixed in 11ae5a1. The new test now uses the repository-required `Copyright 2025-present` header. Verified with `./mvnw test -pl spring-data-valkey -Dtest=ValkeyGlideKeyCommandsUnitTests` (1 test, 0 failures/errors).

**Comment by progys** [comment]
@coderabbitai approve

**Comment by progys** [comment]
@jeremyprime maybe you could please take a look at this and approve? it is a trivial change, we could merge and release it. Thank you

File: spring-data-valkey/src/main/java/io/valkey/springframework/data/valkey/connection/valkeyglide/ValkeyGlideKeyCommands.java [hunk 2]
```diff
@@ -216,20 +221,10 @@ public Cursor.CursorId getId() {
 
 		@Override
 		public boolean hasNext() {
-			// First check if we have items in current batch
-			if (currentBatch != null && currentBatch.hasNext()) {
-				return true;
+			while (!finished && (currentBatch == null || !currentBatch.hasNext())) {
```
Comment (jeremyprime) [inline spring-data-valkey/src/main/java/io/valkey/springframework/data/valkey/connection/valkeyglide/ValkeyGlideKeyCommands.java:224]: Besides this class (and `ValkeyGlideClusterScanCursor` which already supports empty batches), there are others that need a similar fix:
- `ValkeyGlideSetScanCursor` (SSCAN)
- `ValkeyGlideHashScanCursor` (HSCAN)
- `ValkeyGlideZSetScanCursor` (ZSCAN)
- `NodeScopedScanCursor` (routed cluster SCAN)

We should also fix those other commands to correctly handle empty batches.

**Review by jeremyprime — COMMENTED** [review]
@progys Thanks for the contribution!  The fix looks good and tested, we just need to apply the same fix to the other SCAN commands in Valkey GLIDE classes.

**Comment by progys** [comment]
> @progys Thanks for the contribution!  The fix looks good and tested, we just need to apply the same fix to the other SCAN commands in Valkey GLIDE classes.

great, let me do it! thanks for your review!

**Comment by progys** [comment]
@jeremyprime pushed the changes. Hope now all is good.

**Review by jeremyprime — APPROVED** [review]
